### ISSUE TITLE
[BUG] Can't import Harvester into Airgapped Rancher

### ISSUE THREAD

**Issue by noahgildersleeve** [harvester/harvester#2046]
**Describe the bug**

When you try to import Harvester into airgapped rancher it gives an error and never finishes import
**To Reproduce**
1. Setup airgapped rancher with private registry
2. Import Harvester into Rancher

**Expected behavior**

It should finish import
**Support bundle**

[supportbundle_1605e5d6-2c36-418d-8083-b92e0113472e_2022-03-20T22-13-19Z.zip](https://github.com/harvester/harvester/files/8312276/supportbundle_1605e5d6-2c36-418d-8083-b92e0113472e_2022-03-20T22-13-19Z.zip)

**Environment:**
 - Harvester ISO version: v1.0.1-rc1
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Virtualized KVM environment with private network setup and external network
 - Rancher Version: v2.6.4-rc10

**Additional context**
I created the RKE cluster that I did the helm install on with RKE v1.3.8-rc9 so that it matched the cached images needed in v2.6.4-rc10. I also edited the rancher template to only have 1 replicaset and the issuer template to use the proper cert-manager API endpoint. Cert-manager was v1.7.1.
You can see the error that was being generated
```
./logs/harvester-system/harvester-5db7cb56f4-kbcbm/apiserver.log:2022-03-20T22:13:27.922447887Z time="2022-03-20T22:13:27Z" level=error msg="error syncing 'cluster-registration-url': handler harvester-setting-controller: Get \"https://helm-install.local/v3/import/dp6t4gbkvzgpp2bvq9n86hzbg9t95sr6knlsd7wskw4txbdgsljf2c_c-m-lr9hcl2x.yaml\": dial tcp: lookup helm-install.local on 10.53.0.10:53: server misbehaving, requeuing"
```
This was with cert-manager pushing out self-signed certs and the DNS resolution being handled by manually editing `/etc/hosts`
A private registry was setup with 2 NICs. One on the internal network and one on the external network. I was using the walkthrough [here](https://rancher.com/docs/rancher/v2.6/en/installation/other-installation-methods/air-gap/).

**Comment by FrankYang0529** [comment]
I cannot reproduce this with the following steps. My environment is 1-node Harvester `master-a16a5735-head` and Rancher `v2.6.4-rc10`.

### Phase 1: setup airgap Harvester
1. Set `offline: true` in [vagrant-pxe-harvester](https://github.com/harvester/ipxe-examples/blob/f75483563f192090dadd48051bbc3b538c30cd34/vagrant-pxe-harvester/settings.yml#L42-L44).
2. Run `./setup_harvester.sh`.
3. Check we cannot access the internet in Harvester.
```
> ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
^C
--- 1.1.1.1 ping statistics ---
3 packets transmitted, 0 received, 100% packet loss, time 2042ms
```

### Phase 2: setup Rancher and preload images (with Network)
1. Run a Ubuntu VM with two network interfaces. One is `harvester` and another one is `vagrant-libvirt`, so the VM can connect to the internet for initialization.
2. Install docker for running Rancher and private registry.
3. Install skopeo to copy images to a private registry.
```
sudo apt-get -y update
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_${VERSION_ID}/Release.key | sudo apt-key add -
sudo apt-get update
sudo apt-get -y install skopeo
```
4. Run Rancher.
```
docker run --privileged --name rancher -d --restart=unless-stopped -p 80:80 -p 443:443 rancher/rancher:v2.6.4-rc10
```
5. Run private registry.
```
docker run -d -p 5000:5000 --restart=always --name private-registry registry:2
```
6. Copy images to the private registry. Remember to change IP to your private IP.
```
skopeo copy --dest-tls-verify=false docker://docker.io/rancher/rancher-agent:v2.6.4-rc10 docker://192.168.0.50:5000/rancher/rancher-agent:v2.6.4-rc10
```
7. Remove `vagrant-libvirt` interface for the VM, so Rancher will be in air gap environment.
8. Check we cannot access the internet in Rancher VM.
```
> ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1) 56(84) bytes of data.
^C
--- 1.1.1.1 ping statistics ---
1 packets transmitted, 0 received, 100% packet loss, time 0ms
```

### Phase 3: Change registry in Harvester and import it to Rancher
1. Create a file under path `/etc/rancher/rke2/registries.yaml` in Harvester with the following content. Remember to change IP to your private IP.
```
mirrors:
  docker.io:
    endpoint:
    - "http://192.168.0.50:5000"
```
2. Restart RKE2 in Harvester.
```
> systemctl restart rke2-server.service
```
3. Import Harvester to Rancher.
![Screenshot from 2022-03-22 17-26-54](https://user-images.githubusercontent.com/4927361/159449302-d145fefc-2564-43b1-96b9-df9c52a12ac3.png)
![Screenshot from 2022-03-22 17-27-15](https://user-images.githubusercontent.com/4927361/159449380-cc61d7e0-f8cd-4116-899a-cc6c82172178.png)

**Comment by noahgildersleeve** [comment]
That is similar to how I originally tested this, but it's not a representative test of the functionality. Here is the walkthrough that goes over these steps https://rancher.com/docs/rancher/v2.6/en/installation/other-installation-methods/air-gap/ . 

1. Set up a private registry server 
2. Get a copy of every image that is necessary for Rancher to be installed via helm (there's a script in the walkthrough)
3. Upload those to the private registry (there's a script in the walkthrough
4. Set up a Kubernetes cluster to install Rancher on that is not connected to the internet. The easiest way to do this is to have a machine that has two NICs (one on the private network and one on an external network)
5. Do a helm install of Rancher on the k8s cluster while pointing at the private registry in the config (I used RKE for this)
6. Setup Harvester in the private network
7. Import Harvester into the Rancher cluster that is now air gapped

I have some notes and some of the configs I used for testing this if you need them. Keep in mind that you need to have a DHCP server in the private network.

**Comment by alexdepalex** [comment]
@joeyloman Seems like todays adventure.

**Comment by joeyloman** [comment]
Yes indeed. We get the "Unexpected end of JSON input" error and it stays on provisioning state (see screenshot) when there is a container registry configured in the system-default-registry global setting. When I remove the setting value, the Harvester clusters become Active/happy again:

![rancher](https://user-images.githubusercontent.com/4496390/159854454-3ed7db08-1c67-4c7a-834f-d69a65783ded.png)

Could this be related?

**Comment by FrankYang0529** [comment]
I can reproduce the error like @joeyloman mentioned. Following are my steps to reproduce the error.

### Phase 1: setup airgap Harvester

1. Set `offline: true` in [vagrant-pxe-harvester](https://github.com/harvester/ipxe-examples/blob/f75483563f192090dadd48051bbc3b538c30cd34/vagrant-pxe-harvester/settings.yml#L42-L44).
2. Run `./setup_harvester.sh`.

### Phase 2: Setup Rancher from a private registry

1. Create a VM with two NIC and 250G storage. One NIC is to connect to internet. Another is to connect to Harvester.
2. Install Docker and helm.
3. Create folders.
```
mkdir certs
mkdir auth
mkdir registry
sudo mkdir -p /etc/docker/certs.d/myregistry.local:5000
```
4. Generate SSL.
```
openssl req \
  -newkey rsa:4096 -nodes -sha256 -keyout certs/domain.key \
  -addext "subjectAltName = DNS:myregistry.local" \
  -x509 -days 365 -out certs/domain.crt
```
5. Run private registry.
```
docker run -d \
  -p 5000:5000 \
  --restart=always \
  --name registry \
  -v "$(pwd)"/certs:/certs \
  -v "$(pwd)"/registry:/var/lib/registry \
  -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
  -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
  registry:2
```
6. Edit DNS record.
```
# vim /etc/hosts
192.168.0.51 myregistry.local
```
7. Get Rancher scripts.
```
wget https://github.com/rancher/rancher/releases/download/v2.6.4-rc10/rancher-images.txt
wget https://github.com/rancher/rancher/releases/download/v2.6.4-rc10/rancher-load-images.sh
wget https://github.com/rancher/rancher/releases/download/v2.6.4-rc10/rancher-save-images.sh
chmod +x ./rancher-save-images.sh
chmod +x ./rancher-load-images.sh
```
8. Add cert-manager to `rancher-images.txt`
```
helm repo add jetstack https://charts.jetstack.io
helm repo update
helm fetch jetstack/cert-manager --version v1.7.1
helm template ./cert-manager-v1.7.1.tgz | awk '$1 ~ /image:/ {print $2}' | sed s/\"//g >> ./rancher-images.txt
```
9. Pull images. (This step may take 1 ~ 2 hours)
```
sort -u rancher-images.txt -o rancher-images.txt
./rancher-save-images.sh --image-list ./rancher-images.txt
```
10. Copy SSL.
```
sudo cp certs/domain.crt /etc/docker/certs.d/myregistry.local:5000/domain.crt
```
11. Load images to private registry.
```
./rancher-load-images.sh --image-list ./rancher-images.txt --registry myregistry.local:5000
```
12. Run rancher with private registry.
```
docker run -d --restart=unless-stopped \
    -p 80:80 -p 443:443 \
    -e CATTLE_SYSTEM_DEFAULT_REGISTRY=myregistry.local:5000 \
    -e CATTLE_SYSTEM_CATALOG=bundled \
    --privileged \
    --name rancher \
    myregistry.local:5000/rancher/rancher:v2.6.4-rc10
```

### Phase 3: Integrate Rancher and Harvester
1. Add `myregistry.local` DNS record to Harvester
```
# vim /etc/hosts
192.168.0.51 myregistry.local
```
2. Add `registries.yaml` to Harvester
```
# vim /etc/rancher/rke2/registries.yaml
mirrors:
  docker.io:
    endpoint:
      - "https://myregistry.local:5000"
configs:
  "myregistry.local:5000":
    tls:
      insecure_skip_verify: true
```
3. Import Harvester to Rancher.
![image](https://user-images.githubusercontent.com/4927361/159906967-4c7f4505-dd4b-44e1-a3cd-39ca239ea976.png)

**Comment by FrankYang0529** [comment]
@noahgildersleeve, this issue has been fixed in Rancher v2.6.4-rc13 and I can't reproduce it, so I move to Ready for testing.

**Comment by TachunLin** [comment]
Verified **fixed** on Rancher `v2.6.4-rc13` with Harvester `v1.0.1-rc1`. 

### Result
Now can import Harvester to Rancher when both of them are under Airgapped environment (with no proxy server)
![image](https://user-images.githubusercontent.com/29251855/160383219-4e13cd46-35f4-4a79-acd6-ccb9bcf87662.png)



* The airgapped environment setup include the following phases:
1. Prepare the air gapped harvester 
2. Create a virtual machine for Rancher in air gapped environment 
3. Setup a private docker registry and download all the offline images for Rancher 
4. Push the image to docker registry 
5. Install K3s and Rancher 
6. Edit Rancher vm hosts file and boostrap Rancher
7. Edit Harvester cluster hosts files
8. Add host mapping to coredns configmap and deployments 
9. Import Harvester from Rancher UI

### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: v1.0.1-rc1
* Rancher version: v2.6.4-rc13

### Environment setup

#### Phase 1: setup airgap Harvester
1. Set offline: true in [vagrant-pxe-harvester](https://github.com/harvester/ipxe-examples/blob/f75483563f192090dadd48051bbc3b538c30cd34/vagrant-pxe-harvester/settings.yml#L42-L44).
2. Set 1 node harvester with sufficient resources
3. Run ./setup_harvester.sh.

#### Phase 2: Create a virtual machine for Rancher in air gapped environment 
1. Create a virtual machine with at least 300GB storage (OS: Ubuntu Desktop 20.04)
2. Add `harvester` and `vagrant-libvirt` network to the virtual machine  
  - `harvester` for internal 
  - `vagrant-libvirt` for external 

#### Phase 3: Setup a private docker registry and download all the offline images for Rancher 
1. Install VIM
  ```
  sudo apt update
  sudo apt install vim
  ```
2. Install Docker 
  ```
  sudo apt-get update
  sudo apt-get install \
      ca-certificates \
      curl \
      gnupg \
      lsb-release
  curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
  echo \
    "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
    $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
  sudo apt-get update
  sudo apt-get install docker-ce docker-ce-cli containerd.io
  ```
3. Add current user to docker group
  ```
  sudo groupadd docker
  sudo usermod -aG docker $USER
  ```
4. Logout and login again.
5. Install helm
  ```
  curl https://baltocdn.com/helm/signing.asc | sudo apt-key add -
  sudo apt-get install apt-transport-https --yes
  echo "deb https://baltocdn.com/helm/stable/debian/ all main" | sudo tee /etc/apt/sources.list.d/helm-stable-debian.list
  sudo apt-get update
  sudo apt-get install helm
  ```
6. Create certs folder
  ```
  mkdir -p certs
  ```
7. Generate private registry certificate files
  ```
  openssl req \
    -newkey rsa:4096 -nodes -sha256 -keyout certs/domain.key \
    -addext "subjectAltName = DNS:myregistry.local" \
    -x509 -days 365 -out certs/domain.crt
  ```
8. Move certificate files
  ```
  sudo mkdir -p /etc/docker/certs.d/myregistry.local:5000
  sudo cp certs/domain.crt /etc/docker/certs.d/myregistry.local:5000/domain.crt
  ```
9. Start docker registry
  ```
  docker run -d \
    -p 5000:5000 \
    --restart=always \
    --name registry \
    -v "$(pwd)"/certs:/certs \
    -v "$(pwd)"/registry:/var/lib/registry \
    -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
    -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
    registry:2
  ```
10. Add myregistry.local record. Remember to change your private ip.
  ```
  # vim /etc/hosts
  192.168.0.50 myregistry.local
  ```
11. Create `get-rancher-scripts` script.
  ```
  # vim get-rancher-scripts
  #!/bin/bash
  if [[ $# -eq 0 ]] ; then
      echo 'This requires you to pass a version for the url like "v2.6.3"'
      exit 1
  fi
  wget https://github.com/rancher/rancher/releases/download/$1/rancher-images.txt
  wget https://github.com/rancher/rancher/releases/download/$1/rancher-load-images.sh
  wget https://github.com/rancher/rancher/releases/download/$1/rancher-save-images.sh
  chmod +x ./rancher-save-images.sh
  chmod +x ./rancher-load-images.sh
  ```
12. Make `get-rancher-scripts` can be executed.
  ```
  chmod +x get-rancher-scripts
  ```
13. Get  rancher-images.txt.
  ```
  ./get-rancher-scripts v2.6.4-rc13
  ```
14. Add cert-manager images to rancher-images.txt
  ```
  helm repo add jetstack https://charts.jetstack.io/
  helm repo update
  helm fetch jetstack/cert-manager --version v1.7.1
  helm template ./cert-manager-v1.7.1.tgz | awk '$1 ~ /image:/ {print $2}' | sed s/\"//g >> ./rancher-images.txt
  ```
15. Sort rancher-images.txt
  ```
  sort -u rancher-images.txt -o rancher-images.txt
  ```
16. Get Rancher images. This step may take at least 2 hours depends on your network speed
  ```
  ./rancher-save-images.sh --image-list ./rancher-images.txt
  ```
  ![image](https://user-images.githubusercontent.com/29251855/160348531-f6f5dd3b-af01-43e2-81ae-ab5b47719d7f.png)
  
#### Phase 4: Push the image to docker registry 
1. Download k3s v1.23.4+k3s1
  ```
  wget https://github.com/k3s-io/k3s/releases/download/v1.23.4%2Bk3s1/k3s-airgap-images-amd64.tar
  wget https://github.com/k3s-io/k3s/releases/download/v1.23.4%2Bk3s1/k3s
  chmod +x k3s
  
  sudo cp k3s /usr/local/bin/
  sudo chown $USER /usr/local/bin/k3s
  
  curl https://get.k3s.io/ -o install.sh
  chmod +x install.sh
  ```
2. Download rancher v2.6,4-rc13 helm
  ```
  helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
  helm fetch rancher-latest/rancher --version=v2.6.4-rc13
  ```
3. Download cert-manager crds.
  ```
  mkdir cert-manager
  curl -L -o cert-manager/cert-manager-crd.yaml https://github.com/jetstack/cert-manager/releases/download/v1.7.1/cert-manager.crds.yaml
  ```

4. Install k9s (option)
  ```
  curl -kL https://github.com/derailed/k9s/releases/download/v0.25.18/k9s_Linux_x86_64.tar.gz > k9s.tar.gz
  tar -zxvf k9s.tar.gz
  mv k9s /usr/local/bin/
  ```

5. Cut Network -> Remove the `vagrant-libvirt` network device from virtual machine, make it air gapped
6. Load Rancher images to private registry. 
  ```
  ./rancher-load-images.sh --image-list ./rancher-images.txt --registry myregistry.local:5000
  ```
  - This step may take 50 minutes depends on total image size

#### Phase 5: Install K3s and Rancher 
1. Move k3s images files.
  ```
  sudo mkdir -p /var/lib/rancher/k3s/agent/images/
  sudo cp ./k3s-airgap-images-amd64.tar /var/lib/rancher/k3s/agent/images/
  ```
2. Install k3s
  ```
  INSTALL_K3S_SKIP_DOWNLOAD=true ./install.sh
  ```
3. Add `registries.yaml`  to `/etc/rancher/k3s/`
  ```
  # vim /etc/rancher/k3s/registries.yaml
  mirrors:
    docker.io:
      endpoint:
        - "https://myregistry.local:5000/"
  configs:
    "myregistry.local:5000":
      tls:
        insecure_skip_verify: true
  ```
4. Restart k3s
  ```
  sudo systemctl restart k3s.service
  ```
5. Copy kubeconfig to ~/.kube
  ```
  mkdir ~/.kube
  sudo cp /etc/rancher/k3s/k3s.yaml ~/.kube/config
  sudo chown $USER ~/.kube/config
  export KUBECONFIG=~/.kube/config
  ```
6. Generate cert-manager YAML files.
  ```
  helm template cert-manager ./cert-manager-v1.7.1.tgz --output-dir . \
      --namespace cert-manager \
      --set image.repository=myregistry.local:5000/quay.io/jetstack/cert-manager-controller \
      --set webhook.image.repository=myregistry.local:5000/quay.io/jetstack/cert-manager-webhook \
      --set cainjector.image.repository=myregistry.local:5000/quay.io/jetstack/cert-manager-cainjector \
      --set startupapicheck.image.repository=myregistry.local:5000/quay.io/jetstack/cert-manager-ctl
  ```
7. Install cert-manager.
  ```
  kubectl create namespace cert-manager
  kubectl apply -f cert-manager/cert-manager-crd.yaml
  kubectl apply -R -f ./cert-manager
  ```
8. Create `create-self-signed-cert.sh`, copy the content in the [signed certificate guide ](http://docs.rancher.cn/docs/rancher2.5/installation/resources/advanced/self-signed-ssl/_index/#4-%E5%A6%82%E4%BD%95%E7%94%9F%E6%88%90%E8%87%AA%E7%AD%BE%E5%90%8D%E8%AF%81%E4%B9%A6)
9. Make create-self-signed-cert.sh be excutable.
  ```
  chmod +x create-self-signed-cert.sh
  ```
10. Create certificate files for helm-install.local . Remember to change your private IP.
  ```
  ./create-self-signed-cert.sh --ssl-domain=helm-install.local --ssl-trusted-ip=192.168.0.50
  ```
11. Create cattle-system namespace
  ```
  kubectl create ns cattle-system
  ```
12. Create tls-sa secret.
  ```
  kubectl -n cattle-system create secret generic tls-ca \
    --from-file=cacerts.pem=./cacerts.pem
  ```
13. Create tls-rancher-ingress secret.
  ```
  kubectl -n cattle-system create secret tls tls-rancher-ingress \
    --cert=tls.crt \
    --key=tls.key
  ```
14. Generate Rancher YAML files.
  ```
  helm template rancher ./rancher-2.6.4-rc13.tgz --output-dir . \
      --no-hooks \
      --namespace cattle-system \
      --set hostname=helm-install.local \
      --set rancherImageTag=v2.6.4-rc13 \
      --set rancherImage=myregistry.local:5000/rancher/rancher \
      --set systemDefaultRegistry=myregistry.local:5000 \
      --set useBundledSystemChart=true \
      --set ingress.tls.source=secret \
      --set privateCA=true
  ```
15. Install Rancher 
  ```
  kubectl -n cattle-system apply -R -f ./rancher
  
  davidtclin@davidtclin-Standard-PC-Q35-ICH9-2009:~$ kubectl -n cattle-system apply -R -f ./rancher
  clusterrolebinding.rbac.authorization.k8s.io/rancher created
  deployment.apps/rancher created
  ingress.networking.k8s.io/rancher created
  service/rancher created 
  serviceaccount/rancher created
  
  ```
  ![image](https://user-images.githubusercontent.com/29251855/160362486-4e521a71-9b31-4460-a4a0-958653dbd56d.png)

#### Phase 6: Edit Rancher vm hosts file and boostrap Rancher
1. Edit the `/etc/hosts` file in Rancher virtual machine 
  ```
  192.168.0.50    myregistry.local
  ```
  ![image](https://user-images.githubusercontent.com/29251855/160388430-d91b7a6a-8f6b-486a-a07d-9f13dc5e8ece.png)

2. Edit the `/etc/hosts` file in Host machine
  ```
  192.168.0.50 helm-install.local
  ```
  ![image](https://user-images.githubusercontent.com/29251855/160389038-40cf1803-4914-4dae-83a7-0a547872429f.png)

3. Open browser on your host machine, access the URL https://helm-install.local/ 
  ![image](https://user-images.githubusercontent.com/29251855/160363090-3741cb41-a48e-4512-90ab-0ef2287e27ae.png)
 
4. Copy the the helm command and run in the rancher virtual machine terminal 

5. Get the password and complete the bootstrap process
   ![image](https://user-images.githubusercontent.com/29251855/160363211-980d17a8-8441-45ee-9af2-f4d000ac9c06.png)

#### Phase 7: Edit Harvester cluster hosts files
1. Edit `/etc/hosts` on harvester node machine 
  ```
  # vim /etc/hosts
  192.168.0.50 myregistry.local helm-install.local 
  ```
  ![image](https://user-images.githubusercontent.com/29251855/160390326-d89b5ece-2cab-48bc-a1ee-35fd7f8cc648.png)

2. Set registries.yaml in Harvester. /etc/rancher/rke2/registries.yaml
  ```
  # vim /etc/rancher/rke2/registries.yaml
  mirrors:
    docker.io:
      endpoint:
        - "https://myregistry.local:5000/"
  configs:
    "myregistry.local:5000":
      tls:
        insecure_skip_verify: true
  ```
3. Restart Harvester
  ```
  systemctl restart rke2-server.service
  ```

#### Phase 8: Add host mapping to coredns configmap and deployments 
1. Open K9s -> enter `: configmap` -> search dns
  ![image](https://user-images.githubusercontent.com/29251855/160373835-c4aa114e-d07d-4ec6-8ff7-ba742008bf9a.png)
2. Edit the `rke2-coredns-rke2-coredns`
3. Add the following content and save 
  ```
  hosts /etc/coredns/customdomains.db helm-install.local {\n
      \   fallthrough\n    }
  ```

  ```
  customdomains.db: |
      192.168.0.50 helm-install.local
  ```
  
  ```
  data:
    Corefile: ".:53 {\n    errors \n    health  {\n        lameduck 5s\n    }\n    ready
      \n    kubernetes   cluster.local  cluster.local in-addr.arpa ip6.arpa {\n        pods
      insecure\n        fallthrough in-addr.arpa ip6.arpa\n        ttl 30\n    }\n    prometheus
      \  0.0.0.0:9153\n   hosts /etc/coredns/customdomains.db helm-install.local {\n
      \   fallthrough\n    }\n forward   . /etc/resolv.conf\n    cache   30\n    loop
      \n    reload \n    loadbalance \n}"
    customdomains.db: |
      192.168.0.50 helm-install.local
  ```
  
4.  Open K9s -> enter `: deployment` -> search dns 
5.  Edit the rke2-coredns-rke2-coredns 
  ![image](https://user-images.githubusercontent.com/29251855/160374673-cadbf036-7a76-48c3-8029-306976f69ef0.png)
6.  Add the following content and save
  ```
  - key: customdomains.db
              path: customdomains.db
  ```
  ![image](https://user-images.githubusercontent.com/29251855/160374909-e8364442-9401-468f-8c3a-5f7ce9c1443b.png)


### Verify Steps
1. Environment preparation as above steps
2. Open Rancher virtualization management page and click import 
  ![image](https://user-images.githubusercontent.com/29251855/160376484-8d4503ab-32e3-4579-bb54-9f317763e795.png)

3. Copy the registration URL and paste in the Harvester cluster URL settings
  ![image](https://user-images.githubusercontent.com/29251855/160376541-8ad0a58b-e118-4748-9e2d-f0a99eeaca2a.png)

**Comment by TachunLin** [comment]
After discussion. Close this issue. The follow up RKE2 provisioning will track in #1808
